opus_codec_registration.src.html

```diff
@@ -151,6 +159,24 @@
   <dd>
     Configures the frame duration, in milliseconds, of output {{EncodedAudioChunk}}s.
   </dd>
+  <dt><dfn dict-member for=OpusEncoderConfig>complexity</dfn></dt>
+  <dd>
+    Configures the encoder's computational complexity. The valid range is `0` to `10`,
+    with `10` representing the highest complexity.
+  </dd>
+  <dt><dfn dict-member for=OpusEncoderConfig>packetlossperc</dfn></dt>
+  <dd>
+    Configures the encoder's expected packet loss percentage. The valid range is
+    `0` to `100`.
+  </dd>
+  <dt><dfn dict-member for=OpusEncoderConfig>useinbandfec</dfn></dt>
+  <dd>
+    Configures whether the encoder should generate Opus in-band fec or not.
```

Comment:
FEC. Is there a link to the parameter definitions in the opus spec that we can use for each of these?

Reply:
I think we can link to section 6.1 of RFC7587 for useinbandfec and usedtx parameters. We could additionally link to [section 3.2](https://datatracker.ietf.org/doc/html/rfc7587#section-3.2) for complexity. There packetlossperc isn't defined in RFC7587, so it might be best to link to ORTC for all parameters?

I'm not sure what the "hierarchy" of specs is, and if we can mix & match.

Reply:
RFC 7587 Section 6.1 defines `useinbandfec` and `usedtx`  in terms of decoder preference rather than an encoder setting. So linking to Section 6.1 might be a bit confusing.   You might define them like this: 

useinbandfec:  specifies whether the encoder provides Opus in-band FEC.  Possible values are 1 (yes) and 0 (no). 
If no value is specified, useinbandfec is assumed to be 0. 

usedtx:  specifies if the encoder uses DTX.  Possible values are 1 (yes) and 0 (no). If no value is specified, the default is 0.

Reply:
packetlossperc: the packet loss perc parameter could be changed during encoding, this need the OpusEncoderConfig support realime configuration, do we need add describe about this?

Reply:
There have been some questions relating to re-configuration so it might be a good idea to clarify that implementations are required to support this. However, the ability to re-configure encoders is not specific to Opus or `packetlossperc`.

Reply:
> There have been some questions relating to re-configuration so it might be a good idea to clarify that implementations are required to support this. However, the ability to re-configure encoders is not specific to Opus or `packetlossperc`.

agree,  `packetlossperc` should be 0 by default, and user can re-configure it during encoding  in accordance with uplink audio loss rate（ask encoder to generate inbandFec packet)

Reply:
`complexity`: the default value are different in pc/mobile platform to save encoder complexity, webrtc use 5 in Android, iOS and/or ARM device, others is 9 by default.

Reply:
> [...] support realime configuration [...]
> [...] the ability to re-configure encoders [...]

Is `flush()` followed by `configure()` acceptable here? Or is this about having the ability to only update the `packetlossperc` parameter mid-stream without flushing?

Reply:
> [...] the default value are different in pc/mobile platform [...]

I'm not sure how to specify this in spec-text. I've added tentative text, but I don't think that "mobile platforms" is defined normatively.

I don't know how to specify 2 defaults in the dictionary either.

Reply:
> Possible values are 1 (yes) and 0 (no)

The rest of the WebCodec spec uses `true` and `false` directly, so I am favoring these over `1` and `0`.

Reply:
That makes sense.

Reply:
> > [...] the default value are different in pc/mobile platform [...]
> 
> I'm not sure how to specify this in spec-text. I've added tentative text, but I don't think that "mobile platforms" is defined normatively.
> 
> I don't know how to specify 2 defaults in the dictionary either.



> That makes sense.



> > [...] support realime configuration [...]
> > [...] the ability to re-configure encoders [...]
> 
> Is `flush()` followed by `configure()` acceptable here? Or is this about having the ability to only update the `packetlossperc` parameter mid-stream without flushing?

it should be  only update the `packetlossperc` parameter, not flush.

Reply:
> > [...] the default value are different in pc/mobile platform [...]
> 
> I'm not sure how to specify this in spec-text. I've added tentative text, but I don't think that "mobile platforms" is defined normatively.
> 
> I don't know how to specify 2 defaults in the dictionary either.

add this part as recommendation is more suitable, i have seen your latest update, that make sence 👍 .

Reply:
Are you still planning to linkify FEC and DTX per this conversation?

Reply:
I lost track of the original comment. I've added links to RFC6716 (matching the frameDuration link added in #602)